### ISSUE TITLE
Issue uploading multiple cloudinary images via API (Angular SPA)

### ISSUE THREAD

**Issue by sgireddy** [keystonejs/keystone#891]
Hello friends,

I am working on a SPA application integrating Keystone.js with cloudinary-angular. The problem is that cloudinary-angular is NOT using direct-upload technique, so keystone is ignoring images despite the fact that I am sending JSON data in exact same format that keystone expects (tried this after sending as is didn't work).  

This is a non-issue when uploading single image (heroImage), not sure why it works for single image (type: cloudinaryImage) but not multiple images (type: cloudinaryImages).

Is there any workaround/way to fix this issue? Any hints on how can I use direct-upload (or whatever keystone likes) in API/SPA scenario?

Here is the code on git: https://github.com/sgireddy/KeystoneAngularSPA

Looks like this is what I am hitting in cloudinaryimages.js:

```javascript
/**
 * Updates the value for this field in the item from a data object
 *
 * @api public
 */

cloudinaryimages.prototype.updateItem = function(item, data) {
  // TODO - direct updating of data (not via upload)
};
```

I tried messing around with cloudinaryimages.prototype.getRequestHandler and found that corresponding admin UI code here:

```javascript
// Upload References (direct uploading)
if (req.body[paths.uploads]) {
  var uploads = JSON.parse(req.body[paths.uploads]);

  uploads.forEach(function(file) {
    item.get(field.path).push(file);
  });
}
```

whatever req.body[paths.uploads] is actually req.body.images in my code, so I added the following code after above section:

```javascript
if (req.body.images_upload) {
  var uploads = req.body.images_upload; 
  uploads.forEach(function (file) {
    item.get(field.path).push(file);
  });
} 
```

This time I got a new issue:

```
 { [CastError: Cast to ObjectId failed for value "{"public_id":"preset_folder/kgs ******  at path "_id"',
```

Looks like I need to generate mogodb _id for each image, being a node novice I have no idea how to do that either.... but all this is a big hack anyways...

can somebody help, please.....

Thanks,
shashi

**Comment by sgireddy** [comment]
I was able to save images by bypassing update handler but admin UI is not showing the images, instead showing two blank placeholders...




exports.create = function (req, res) {	
 
	var item = new Movie.model(),
		data = (req.method == 'POST') ? req.body : req.query;    
	var movie = new Movie.model();
	
	movie.title = data.title;
	movie.director = data.director;
	movie.releaseYear = data.releaseYear;
	movie.genre = data.genre;
	
	_.each(data.images, function (elm, idx, arr) {
		var tmp = {};
		tmp["public_id"] = elm.public_id; 
		tmp["version"] = elm.version;
		tmp["signature"] = elm.signature;
		tmp["width"] = elm.width;
		tmp["height"] = elm.height;
		tmp["format"] = elm.format;
		tmp["resource_type"] = elm.resource_type;
		tmp["url"] = elm.url;
		tmp["secure_url"] = elm.secure_url;
		tmp["_id"] = mongoose.Types.ObjectId();        
		movie.images.push(tmp);
	});	
	
	movie.save(function (err) {
		if (err) return res.apiError('error', err);
		res.apiResponse(movie);		
	});
}

**Comment by webteckie** [comment]
@JedWatson or anyone familiar with it can someone handle the CloudinaryImages field's updateItem?  I have tried but got a bit confused :-)  I had done the CloudinaryImage a while back but this one seems to be a bit trickier since the path is an array.  Somehow I suspect that the update handler is not passing the correct or sufficient data for updateItem to do its job.  But I could be wrong!

**Comment by porschuetz** [comment]
I never saw the multi images upload work in the Admin UI.

is there any info to why? 

Thanks

**Comment by mxstbr** [comment]
This is getting fixed in 0.4, see #2566 and related issues.

**Comment by intrnet** [comment]
This is an answear: 
```
<label>
	<span>Images</span>
	<input type="hidden" name="images" value="upload:Img">
        <input type="file" accept="image/*,application/pdf,application/postscript" multiple="" name="Img">
</label>
```
